migration-guide.md

```diff
@@ -0,0 +1,500 @@
+indexingPolicy.setIndexingMode(IndexingMode.CONSISTENT); 
+
+// Included paths
+List<IncludedPath> includedPaths = new ArrayList<>();
+IncludedPath includedPath = new IncludedPath();
+includedPath.setPath("/*");
+includedPaths.add(includedPath);
+indexingPolicy.setIncludedPaths(includedPaths);
+
+// Excluded paths
+List<ExcludedPath> excludedPaths = new ArrayList<>();
+ExcludedPath excludedPath = new ExcludedPath();
+excludedPath.setPath("/name/*");
+excludedPaths.add(excludedPath);
+indexingPolicy.setExcludedPaths(excludedPaths);
+
+containerProperties.setIndexingPolicy(indexingPolicy);
+
+CosmosAsyncContainer containerIfNotExists = database.createContainerIfNotExists(containerProperties, 400)
+                                                    .block()
+                                                    .getContainer();
+```
+
+**Java SDK 3.x.x Async API:**
+
+```java
+CosmosContainerProperties containerProperties = new CosmosContainerProperties(containerName, "/lastName");
+
+// Custom indexing policy
+IndexingPolicy indexingPolicy = new IndexingPolicy();
+indexingPolicy.setIndexingMode(IndexingMode.CONSISTENT); //To turn indexing off set IndexingMode.NONE
+
+// Included paths
+List<IncludedPath> includedPaths = new ArrayList<>();
+IncludedPath includedPath = new IncludedPath();
+includedPath.path("/*");
+includedPaths.add(includedPath);
+indexingPolicy.setIncludedPaths(includedPaths);
+
+// Excluded paths
+List<ExcludedPath> excludedPaths = new ArrayList<>();
+ExcludedPath excludedPath = new ExcludedPath();
+excludedPath.path("/name/*");
+excludedPaths.add(excludedPath);
+indexingPolicy.excludedPaths(excludedPaths);
+
+containerProperties.indexingPolicy(indexingPolicy);
+
+CosmosContainer containerIfNotExists = database.createContainerIfNotExists(containerProperties, 400)
+                                               .block()
+                                               .container();
+```
+
+### Stored procedures
+
+**Java SDK 4.0 Async API:**
+
+```java
+logger.info("Creating stored procedure...\n");
+
+sprocId = "createMyDocument";
+String sprocBody = "function createMyDocument() {\n" +
+        "var documentToCreate = {\"id\":\"test_doc\"}\n" +
+        "var context = getContext();\n" +
+        "var collection = context.getCollection();\n" +
+        "var accepted = collection.createDocument(collection.getSelfLink(), documentToCreate,\n" +
+        "    function (err, documentCreated) {\n" +
+        "if (err) throw new Error('Error' + err.message);\n" +
+        "context.getResponse().setBody(documentCreated.id)\n" +
+        "});\n" +
+        "if (!accepted) return;\n" +
+        "}";
+CosmosStoredProcedureProperties storedProcedureDef = new CosmosStoredProcedureProperties(sprocId, sprocBody);
+container.getScripts()
+        .createStoredProcedure(storedProcedureDef,
+                new CosmosStoredProcedureRequestOptions()).block();
+
+// ...
+
+logger.info(String.format("Executing stored procedure %s...\n\n", sprocId));
```

Comment:
String.format is unnecessary. please use slf4j placeholder instead.
http://www.slf4j.org/manual.html#typical_usage